junit-jupiter-api/src/main/java/org/junit/jupiter/api/DynamicNode.java

```diff
@@ -40,6 +45,29 @@ public String getDisplayName() {
 		return this.displayName;
 	}
 
+	/**
+	 * Get the optional test source of this {@code DynamicNode}.
+	 *
+	 * @see #withTestSource(TestSource)
+	 */
+	public Optional<TestSource> getTestSource() {
+		return Optional.ofNullable(testSource);
+	}
+
+	DynamicNode setTestSource(TestSource testSource) {
+		this.testSource = testSource;
+		return this;
+	}
+
+	/**
+	 * Create new instance of this dynamic container or test with a custom test source.
```

Comment:
Why would we want to create a new instance?

Why not just return `this` to support method chaining?